src/components/ProgramCollection.js

```diff
@@ -108,151 +134,227 @@ class ProgramCollection extends Component {
+              startIcon={<Add />}
+              component={Link}
+              to="/mission-control"
+            >
+              <FormattedMessage
+                id="app.program_list.new"
+                description="Button label to create new program"
+                defaultMessage="New Program"
               />
-            </Grid.Column>
-            <Grid.Column width={4}>
-              <Header as="h1" textAlign="center">
-                {label}
-              </Header>
-            </Grid.Column>
-            <Grid.Column width={1} />
-            <Grid.Column floated="right" width={2}>
-              <Dropdown
-                text={sortText}
-                icon="sort"
-                floating
-                labeled
-                button
-                className="icon"
-              >
-                <Dropdown.Menu>
-                  <Dropdown.Item onClick={this.handleOrderingChange} name="name">
+            </Button>
+          </Grid>
+        </TitleArea>
+        <Grid container spacing={2}>
+          <Grid item xs={4}>
+            <TextField
+              variant="outlined"
+              size="small"
+              className="prompt"
+              icon="search"
+              placeholder={searchPlaceholder}
+              onChange={this.handleSearchChange}
+              InputProps={{
+                endAdornment: (
+                  <InputAdornment position="end">
+                    <Search />
+                  </InputAdornment>
+                ),
+              }}
+            />
+          </Grid>
+          <Grid item xs={8}>
+            <Grid container justify="flex-end" spacing={2}>
+              <Grid item>
+                <Button aria-controls="sort-menu" aria-haspopup="true" onClick={this.handleSortClick}>
+                  {sortText}
+                </Button>
+                <Menu
+                  id="sort-menu"
+                  anchorEl={sortMenuAnchorElement}
+                  keepMounted
+                  open={Boolean(sortMenuAnchorElement)}
+                  onClose={this.handleSortClose}
+                  getContentAnchorEl={null}
+                  anchorOrigin={{
+                    vertical: 'bottom',
+                    horizontal: 'center',
+                  }}
+                  transformOrigin={{
+                    vertical: 'top',
+                    horizontal: 'center',
+                  }}
+                >
+                  <MenuItem onClick={this.handleNameOrderingChange}>
                     <FormattedMessage
                       id="app.program_collection.name"
                       description="Button label to sort by name"
                       defaultMessage="Name"
                     />
                     {
                       ordering === 'name' ? (
-                        <Icon name="caret down" />
+                        <>
+                          <ArrowDownward />
+                          a ... z
```

Comment:
We probably don't need to handle this now, but this only makes sense in western languages